app/src/main/java/dev/ukanth/ufirewall/Api.java

```diff
@@ -1688,7 +1688,7 @@ private static boolean packagesExistForUserUid(HashMap<Integer,String> pkgs, int
     public static HashMap<Integer, String> getPackagesForUser(List<Integer> userProfile) {
         HashMap<Integer,String> listApps = new HashMap<>();
         for(Integer integer: userProfile) {
-            Shell.Result result = Shell.cmd("pm list packages -U --user " + integer).exec();
+            Shell.Result result = Shell.cmd("pm list packages -u --user " + integer).exec();
```

Comment:
-U is to list the packages with UID. Not sure why you want to remove it ?

Reply:
I was getting consistently an error message returned by "pm list packages -U --user" xx in one of my devices (something similar to "Error: Unknown option: -U").
I believe I was testing the scenario where root is not available (or not supported).

Reply:
better look at `cmd package list packages` as an option for recent implementations/SDK's

Reply:
Getting the application uids is not a problem with the PackageManager APIs, the issue is getting (only) those for a given user: The API for that was only added very recently.

Reply:
Please have a look in the two commits currently in the branch: Those don't solve the underlying problem but already help mitigating it.
As a bonus, those also fix the lag noticed when opening the application list or the log views.